include/mbgl/util/string.hpp

```diff
@@ -10,7 +10,16 @@ namespace util {
 
 template <class T>
 inline std::string toString(T t) {
-    return std::to_string(t);
+    if (t - (int)t > 0.0 || t + (int)t < 0.0) {
```

Comment:
`std::to_string()` can be used with a variety of numeric types. This function ideally shouldn’t restrict the domain of that function at all. The particular bug you’re trying to fix occurs by way of [these lines](https://github.com/mapbox/mapbox-gl-native/blob/5837ba1ad3cd84954711c6d42ccc1358da8a09d3/src/mbgl/renderer/symbol_bucket.cpp#L137-L142), particularly the `double` case. In fact, if you look below here, there are already overrides for `int8_t` and `uint8_t`: I think a proper fix would come in the form of an override for `double`, not a rewrite of this generic function.